webui/react/src/pages/TrialDetails/TrialDetailsLogs.tsx

```diff
@@ -179,26 +212,187 @@ const TrialDetailsLogs: React.FC<Props> = ({ experiment, trial }: Props) => {
     };
   }, [trial?.id, ui.isPageHidden]);
 
+  const onClickSearchIcon = useCallback(() => {
+    searchOn ? setLogViewerOn(false) : setSearchOn(true);
+  }, [searchOn]);
+
   const logFilters = (
     <LogViewerSelect
       options={filterOptions}
-      showSearch={true}
+      searchOn={searchOn}
+      showSearch={false}
       values={filterValues}
       onChange={handleFilterChange}
+      onClickSearch={onClickSearchIcon}
       onReset={handleFilterReset}
     />
   );
 
+  const throttledChangeSearch = useMemo(
+    () =>
+      throttle(
+        500,
+        (s: string) => {
+          updateSettings({ ...settings, searchText: s });
+        },
+        { noLeading: true },
+      ),
+    [updateSettings, settings],
+  );
+
+  useEffect(() => {
+    return () => {
+      throttledChangeSearch.cancel();
+    };
+  }, [throttledChangeSearch]);
+
+  const onSearchChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setSearchInput(e.target.value);
+      throttledChangeSearch(e.target.value);
+      if (!e.target.value) {
+        setSearchResults([]);
+        setSelectedLog(undefined);
+      }
+    },
+    [throttledChangeSearch],
+  );
+
+  const formatedSearchResults = useMemo(() => {
+    const key = settings.searchText;
+
+    if (!key) return;
```

Comment:
we can simplify the code handling `formattedSearchResults` if we flatten the "no key" and "no results" cases:
```suggestion
    if (!key) return [];
```